ARCHITECTURE.md

```diff
@@ -0,0 +1,35 @@
+# PEP Proxy Architecture
```

Comment:
It's a matter of a personal taste, but probably it's better to have filenames in lowercase (architecture.md) taking into account how it's being done at other repos (e.g. Cygnus).

Feel free to ignore this comment with NTC if you don't agree :)